modules/safer-cluster/main.tf

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// The safer-cluster module is based on a private cluster, with a several
+// settings set to recommended values by default.
+module "gke" {
+  source             = "../beta-private-cluster/"
+  project_id         = var.project_id
+  name               = var.name
+  regional           = var.regional
+  region             = var.region
+  network            = var.network
+  network_project_id = var.network_project_id
+
+  // We need to enforce a minimum Kubernetes Version to ensure
+  // that the necessary security features are enabled.
+  kubernetes_version = "latest"
+
+  // Nodes are created with a default version. The nodepool enables
+  // auto_upgrade so that the node versions can be kept up to date with
+  // the master upgrades.
+  //
+  // https://cloud.google.com/kubernetes-engine/versioning-and-upgrades
+  node_version = ""
+
+  master_authorized_networks_config = var.master_authorized_networks_config
+
+  subnetwork        = var.subnetwork
+  ip_range_pods     = var.ip_range_pods
+  ip_range_services = var.ip_range_services
+
+  horizontal_pod_autoscaling = var.horizontal_pod_autoscaling
+  http_load_balancing        = var.http_load_balancing
+
+  // We suggest the use coarse network policies to enforce restrictions in the
+  // communication between pods.
+  //
+  // NOTE: Enabling network policy is not sufficient to enforce restrictions.
+  // NetworkPolicies need to be configured in every namespace. The network
+  // policies should be under the control of a cental cluster management team,
+  // rather than individual teams.
+  network_policy = true
+
+  maintenance_start_time = var.maintenance_start_time
+
+  initial_node_count = var.initial_node_count
+
+  // We suggest removing the default node pull, as it cannot be modified without
+  // destroying the cluster.
+  remove_default_node_pool = true
+
+  node_pools        = var.node_pools
+  node_pools_labels = var.node_pools_labels
+
+  // TODO(mmontan): check whether we need to restrict these
+  // settings.
+  node_pools_metadata = var.node_pools_metadata
+  node_pools_taints   = var.node_pools_taints
+  node_pools_tags     = var.node_pools_tags
+
+  // TODO(mmontan): we generally considered applying
```

Comment:
Let's replace this comment with an issue.

Reply:
Tracked at #319